src/transformer_util.ts

```diff
@@ -0,0 +1,545 @@
+          // with initializers.
+          // find the parent ExpressionStatement like MyClass.foo = ...
+          const expressionStmt =
+              lastNodeWith(nodePath, (node) => node.kind === ts.SyntaxKind.ExpressionStatement);
+          if (expressionStmt) {
+            ts.setSyntheticLeadingComments(
+                expressionStmt, ts.getSyntheticLeadingComments(parent1) || []);
+          }
+        } else if (
+            parent3 && parent3.kind === ts.SyntaxKind.VariableStatement &&
+            tsickle.hasModifierFlag(parent3, ts.ModifierFlags.Export)) {
+          // TypeScript ignore synthetic comments on exported variables.
+          // find the parent ExpressionStatement like exports.foo = ...
+          const expressionStmt =
+              lastNodeWith(nodePath, (node) => node.kind === ts.SyntaxKind.ExpressionStatement);
+          if (expressionStmt) {
+            ts.setSyntheticLeadingComments(
+                expressionStmt, ts.getSyntheticLeadingComments(parent3) || []);
+          }
+        }
+      }
+      // TypeScript ignore synthetic comments on reexport / import statements.
+      const moduleName = extractModuleNameFromRequireVariableStatement(node);
+      if (moduleName && fileContext.importOrReexportDeclarations) {
+        const index = importOrReexportsCount++;
+        const originalImportOrReexportNode = fileContext.importOrReexportDeclarations[index];
+        if (originalImportOrReexportNode) {
+          ts.setSyntheticLeadingComments(
+              node, ts.getSyntheticLeadingComments(originalImportOrReexportNode) || []);
+        }
+      }
+      nodePath.push(node);
+      node.forEachChild(visitNode);
+      nodePath.pop();
+    }
+  };
+}
+
+function extractModuleNameFromRequireVariableStatement(node: ts.Node): string|null {
+  if (node.kind !== ts.SyntaxKind.VariableStatement) {
+    return null;
+  }
+  const varStmt = node as ts.VariableStatement;
+  const decls = varStmt.declarationList.declarations;
+  let init: ts.Expression|undefined;
+  if (decls.length !== 1 || !(init = decls[0].initializer) ||
+      init.kind !== ts.SyntaxKind.CallExpression) {
+    return null;
+  }
+  const callExpr = init as ts.CallExpression;
+  if (callExpr.expression.kind !== ts.SyntaxKind.Identifier ||
+      (callExpr.expression as ts.Identifier).text !== 'require' ||
+      callExpr.arguments.length !== 1) {
+    return null;
+  }
+  const moduleExpr = callExpr.arguments[0];
+  if (moduleExpr.kind !== ts.SyntaxKind.StringLiteral) {
+    return null;
+  }
+  return (moduleExpr as ts.StringLiteral).text;
+}
+
+function lastNodeWith(nodes: ts.Node[], predicate: (node: ts.Node) => boolean): ts.Node|null {
+  for (let i = nodes.length - 1; i >= 0; i--) {
+    const node = nodes[i];
+    if (predicate(node)) {
+      return node;
+    }
+  }
+  return null;
+}
+
+/**
+ * Synthesizes the comments before and after a node
+ * befor calling a callback.
```

Comment:
By default, comments are just text ranges (i.e. from - to). To synthesize means to create a node that is independent of text ranges and contains the information in itself.

For comments, we go from `ts.CommentRange` to `ts.SynthesizedComment`. The latter contains a field `text` to store the actual comment.